weaviate/embedded.py

```diff
@@ -0,0 +1,142 @@
+import os
+import platform
+import signal
+import stat
+import subprocess
+import time
+from dataclasses import dataclass
+import urllib.request
+from pathlib import Path
+import socket
+
+from weaviate.exceptions import WeaviateStartUpError
+
+
+default_binary_path = str((Path.home() / ".cache/weaviate-embedded"))
+default_persistence_data_path = str((Path.home() / ".local/share/weaviate"))
+
+
+@dataclass
+class EmbeddedOptions:
+    persistence_data_path: str = os.environ.get("XDG_DATA_HOME", default_persistence_data_path)
+    binary_path: str = os.environ.get("XDG_CACHE_HOME", default_binary_path)
+    binary_url: str = (
```

Comment:
I had a thought about this parameter name:
I think we will have two different ways of using this.
1) Give a full url to a binary: `"https://github.com/samos123/weaviate/releases/download/v1.17.3/weaviate-server"`
2) Set the version (eg `"1.17.6"`) and we figure out the full url in the client.

The current parameter name only fits to the first option, but not the second. Should we rename this to "weaviate_version" so it fits better to option 2? Maybe you also have a better idea for the parameter name :)

Reply:
(the comment is only about the parameter name, no need to add the aprsing now)

Reply:
I think in majority of people won't bother with setting the version or URL and will use the default. From an end-user perspective, setting the version is a more likely than setting the URL yourself. My current preference is to get more feedback on this feature and then go from there. The binary_url parameter gives the end-user the most flexibility if needed.

I think ideally we would have 2 parameters: `version` and `binary_url` when there are more versions supported by this. Right now only 1.18 has binary release